pkg/actuators/machine/actuator_test.go

```diff
@@ -1281,3 +1277,213 @@ func awsClientBuilderFunc(c awsclient.Client) awsclient.AwsClientBuilderFuncType
 		return c, nil
 	}
 }
+
+func TestPatchMachine(t *testing.T) {
+	// BEGIN: Set up test environment
+	g := NewWithT(t)
+
+	testEnv := &envtest.Environment{
+		CRDDirectoryPaths: []string{filepath.Join("..", "..", "..", "config", "crds")},
+	}
+
+	var err error
+	cfg, err := testEnv.Start()
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(cfg).ToNot(BeNil())
+	defer func() {
+		g.Expect(testEnv.Stop()).To(Succeed())
+	}()
+
+	mgr, err := manager.New(cfg, manager.Options{
+		Scheme:             scheme.Scheme,
+		MetricsBindAddress: "0",
+	})
+	g.Expect(err).ToNot(HaveOccurred())
+
+	actuator, err := initActuator(mgr, t)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	doneMgr := make(chan struct{})
+	go func() {
+		g.Expect(mgr.Start(doneMgr)).To(Succeed())
+	}()
+	defer close(doneMgr)
+
+	// END: setup test environment
+
+	k8sClient := mgr.GetClient()
+
+	awsCredentialsSecret := stubAwsCredentialsSecret()
+	g.Expect(k8sClient.Create(context.TODO(), awsCredentialsSecret)).To(Succeed())
+
+	userDataSecret := stubUserDataSecret()
+	g.Expect(k8sClient.Create(context.TODO(), userDataSecret)).To(Succeed())
+
+	codec, err := providerconfigv1.NewCodec()
+	if err != nil {
+		t.Fatalf("unable to build codec: %v", err)
+	}
+
+	testCases := []struct {
+		name   string
+		mutate func(*machinev1.Machine)
+		expect func(*machinev1.Machine) error
+	}{
+		{
+			name: "Test changing labels",
+			mutate: func(m *machinev1.Machine) {
+				m.ObjectMeta.Labels["testlabel"] = "test"
+			},
+			expect: func(m *machinev1.Machine) error {
+				if m.ObjectMeta.Labels["testlabel"] != "test" {
+					return fmt.Errorf("label \"testlabel\" %q not equal expected \"test\"", m.ObjectMeta.Labels["test"])
+				}
+				return nil
+			},
+		},
+		{
+			name: "Test setting phase",
+			mutate: func(m *machinev1.Machine) {
+				failed := "Failed"
```

Comment:
we can use the constants vendored with the mapi here or create a single variable and reuse it instead of a carrying around a string which is more error prone.
Also you could use `utils.StringPtr` to avoid the extra variable.

Reply:
Mapi constants seem to be private https://github.com/openshift/machine-api-operator/blob/a9085e24cf62862083cfbf295d0e9868b5735a05/pkg/controller/machine/controller.go#L68